### PR TITLE
HTML: remove early return in SVG processing

### PR DESCRIPTION
**Pull request by tuncbkose**
This PR removes some code that was added [a long time ago](https://github.com/shibukawa/sphinx/commit/298ea6ea5be2c169d7dc5bfc0ec1ce9fd524defa) when `docutils` didn't support SVGs. Removing this chunk passed all the tests.

I encountered this when I was trying to use `:height:` and `:width:` options with SVGs. Due to the removed code, if one specifies a unit, like `5cm`, it gets directly passed to the HTML attribute, which is incorrect. With this PR, the size handling is done by `docutils`, which uses the `style` attribute where one can specify units.

One thing to note is that `:scale:` option for SVGs still doesn't work due to https://github.com/sphinx-doc/sphinx/issues/12415. If this PR is accepted as it is, due to not returning early, the use of `:scale:` option for SVGs may emit an additional warning from `docutils`'s `HTMLTranslator::image_size` about `PIL` being required. I'm not sure what is the best way to deal with that.

### Feature or Bugfix
- Bugfix

### Detail
- Fixes specifying units in `:height:` and `:width:` options for SVG files.

### CODE REVIEW

**Comment by picnixz** [comment]
You say that the tests pass but do we actually cover this case? If not, could you add a test covering the use of that option please? (just to see whether docutils is supporting the options correctly?)

**Comment by tuncbkose** [comment]
@picnixz I've added tests checking for the use of the `style` attribute, which is what docutils uses in the HTML.

**Comment by chrisjsewell** [comment]
thanks, 
for reference, this is the upstream visit method: https://github.com/live-clones/docutils/blob/d54b59f99809d2b7d1ff78ff1185452fcb977971/docutils/docutils/writers/_html_base.py#L1145

In addition, to what @picnixz mentioned (understanding if the test cases cover the option use),
I would also just like to clarify here how/where in the code base docutils covers these options

**Comment by tuncbkose** [comment]
In `docutils`'s `visit_image`, the options are obtained [via this method](https://github.com/live-clones/docutils/blob/d54b59f99809d2b7d1ff78ff1185452fcb977971/docutils/docutils/writers/_html_base.py#L1156) (more specifically on [this](https://github.com/live-clones/docutils/blob/d54b59f99809d2b7d1ff78ff1185452fcb977971/docutils/docutils/writers/_html_base.py#L411) line) and put into the `style` attribute right below the first link.

**Review by AA-Turner — APPROVED** [review]
Hi @tuncbkose, please could you add an entry to CHANGES? Otherwise looks good!

A

**Comment by tuncbkose** [comment]
@AA-Turner rebased and added an entry to CHANGES. Thanks for the review!

File: CHANGES.rst [hunk 1]
```diff
@@ -94,6 +94,8 @@ Bugs fixed
 * #11961: Omit anchor references from document title entries in the search index,
   removing duplication of search results.
   Patch by James Addison.
+* #12425: Remove early return in SVG processing for HTML.
```
Comment (AA-Turner) [inline CHANGES.rst]:
```suggestion
* #12425: Use Docutils' SVG processing in the HTML builder and remove internal workaround.
```

File: CHANGES.rst [hunk 2]
```diff
@@ -94,6 +94,8 @@ Bugs fixed
 * #11961: Omit anchor references from document title entries in the search index,
   removing duplication of search results.
   Patch by James Addison.
+* #12425: Use Docutils' SVG processing in the HTML builder and remove internal workaround.
+  Patch by Tunç Başar Köse.
```
Comment (AA-Turner) [inline CHANGES.rst]:
```suggestion
* #12425: Use Docutils' SVG processing in the HTML builder
  and remove Sphinx's custom logic.
  Patch by Tunç Başar Köse.
```

**Pull request merged**